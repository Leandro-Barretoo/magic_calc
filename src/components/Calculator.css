.TableContainer {
  border: 2px solid black;
  width: 500px;
  height: 500px;
  margin: 40px;
  border-collapse: collapse;
}

.TableCell {
  border: 1px solid lightgrey;
  text-align: center;
  font-size: 18px;
  font-weight: 600;
}

.Input {
  text-align: right;
  padding-right: 35px;
  background-color: grey;
  color: white;
  font-weight: 600;
  font-size: 18px;
}

.Orange {
  background-color: lightsalmon;
}

.btn {
  border: none;
  background-color: transparent;
  font-size: 18px;
  font-weight: 600;
  padding: 10px;
}