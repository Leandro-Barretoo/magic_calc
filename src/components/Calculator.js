import React from 'react';
import './Calculator.css';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <table className="TableContainer">
        <tr>
          <td className="TableCell Input" colSpan="4">0</td>
        </tr>
        <tr>
          <td className="TableCell">AC</td>
          <td className="TableCell">+/-</td>
          <td className="TableCell">%</td>
          <td className="TableCell Orange">:</td>
        </tr>
        <tr>
          <td className="TableCell">7</td>
          <td className="TableCell">8</td>
          <td className="TableCell">9</td>
          <td className="TableCell Orange">x</td>
        </tr>
        <tr>
          <td className="TableCell">4</td>
          <td className="TableCell">5</td>
          <td className="TableCell">6</td>
          <td className="TableCell Orange">-</td>
        </tr>
        <tr>
          <td className="TableCell">1</td>
          <td className="TableCell">2</td>
          <td className="TableCell">3</td>
          <td className="TableCell Orange">+</td>
        </tr>
        <tr>
          <td className="TableCell" colSpan="2">0</td>
          <td className="TableCell">.</td>
          <td className="TableCell Orange">=</td>
        </tr>
      </table>
    );
  }
}

export default Calculator;
