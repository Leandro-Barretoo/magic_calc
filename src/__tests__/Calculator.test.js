import React from 'react';
import '@testing-library/jest-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Calculator from '../components/Calculator';

configure({ adapter: new Adapter() });

describe('the UI page', () => {
  test('displays correctly', () => {
    const component = shallow(<Calculator />);
    expect(component).toMatchSnapshot();
  });
});
