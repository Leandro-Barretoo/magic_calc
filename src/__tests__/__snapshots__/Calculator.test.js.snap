// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the UI page displays correctly 1`] = `ShallowWrapper {}`;
